### DISCUSSION TITLE
Applicability of Avalanche library to regression problem?

### DISCUSSION THREAD

**Discussion by highclef** [ContinualAI/avalanche#867] [Q&A] [answered]
Hi all,
I read a post https://github.com/ContinualAI/avalanche/discussions/407 and it seems like Avalanche is suitable for classification problems, not for regression problems. Also the 91th line of base_strategy.py uses `CrossEntropyLoss()`, which is actually for classification : https://github.com/ContinualAI/avalanche/blob/master/avalanche/training/strategies/base_strategy.py

Here are my questions:
1. Can I still use Avalanche for regression problem?
2. I used `MSELoss()` for my regression problem and here is the code:
```python
# define optimizer and loss function for training
optimizer = SGD(model.parameters(), lr=0.001, momentum=0.9)
criterion = MSELoss()

# Instantiate Continual learning strategy
strategy = Naive(
     model, optimizer, criterion, train_mb_size=32, train_epochs=2,
     eval_mb_size=32)
```
Does base_strategy overwrite my criterion and applies `CrossEntropyLoss()` automatically, no matter which criterion I defined, or am I still having `MSELoss()` on my result?
3. According to https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html, `MSELoss()` measures the mean squared error. But what does "Accuracy" mean in this case?

```
{'Loss_Epoch/train_phase/train_stream/Task000': 0.19414528276603987,
  'Loss_Exp/eval_phase/test_stream/Task000/Exp000': 0.24790968985511008,
  'Loss_Exp/eval_phase/test_stream/Task000/Exp001': 0.2299130112751,
  'Loss_Exp/eval_phase/test_stream/Task000/Exp002': 0.2401025802251622,
  'Loss_Stream/eval_phase/test_stream/Task000': 0.23930842711845743,
  'Top1_Acc_Epoch/train_phase/train_stream/Task000': 0.35983479607640684,
  'Top1_Acc_Exp/eval_phase/test_stream/Task000/Exp000': 0.2794386821232459,
  'Top1_Acc_Exp/eval_phase/test_stream/Task000/Exp001': 0.16015863331299574,
  'Top1_Acc_Exp/eval_phase/test_stream/Task000/Exp002': 0.2684563758389262,
  'Top1_Acc_Stream/eval_phase/test_stream/Task000': 0.2360178970917226}
```
It doesn't look like `Acc = 1 - Loss`. What is calculcated for Acc in this case and how should I interpret this result? Should I just accept the losses and ignore accs?

**Comment by AndreaCossu** [comment, marked as answer]
Hi @highclef ! Yes, you can use Avalanche for regression. The `MSELoss` you are passing is used as loss during the strategy execution, so it should work fine: the `BaseStrategy` does not override your loss with `CrossEntropyLoss`.  

Currently, the default logger used by the strategy returns the loss and the accuracy. In your case, you can either discard the accuracy values (since they are not meaningful for your problem) or, better, feed to the strategy your custom `EvaluationPlugin` instance with the metrics you want (take a look at [this tutorial](https://avalanche.continualai.org/from-zero-to-hero-tutorial/06_loggers) if you are not sure how to do this).

Hope this helps!
